src/emulator/apphosting/serve.ts

```diff
@@ -85,9 +85,15 @@ function availablePort(host: string, port: number): Promise<boolean> {
   });
 }
 
-function getEmulatorEnvs(): Record<string, string> {
+/**
+ * Exported for unit tests
+ */
+export function getEmulatorEnvs(): Record<string, string> {
   const envs: Record<string, string> = {};
-  const emulatorInfos = EmulatorRegistry.listRunningWithInfo();
+  // No need to set envs for the apphosting emulator itself.
+  const emulatorInfos = EmulatorRegistry.listRunningWithInfo().filter(
+    (emulator) => emulator.name !== "apphosting",
```

Comment:
should we be using enums instead?

https://github.com/firebase/firebase-tools/blob/5b04f7d375bdd97d806253bb03d2a1988aeae567/src/emulator/types.ts#L11